### ISSUE TITLE
sort_unnamed_constructors_first

### ISSUE THREAD

**Issue by bwilkerson** [dart-lang/linter#3223]
The rule treats constructors named `new` as being named, when in reality `new` is equivalent to being unnamed. I don't know whether the rule wants to be updated. More concretely, the question is, should the following produce a lint:
```dart
class C {
  C.named();
  C.new(); // LINT?
}
```

**Comment by goderbauer** [comment]
I would expected that the `unnecessary_constructor_name` [1] warns you to change `new` to an unnamed constructor, which I think is the right thing to do here. If you explicitly want to name the unnamed constructor I would expect regular sorting rules for names apply (but I also don't care too much about that case as we are going to recommend the `unnecessary_constructor_name` lint [2] in package:lints/package:flutter_lints).

[1] https://dart-lang.github.io/linter/lints/unnecessary_constructor_name.html
[2] https://github.com/dart-lang/lints/issues/68
